src/components/braintree/utils.ts

```diff
@@ -0,0 +1,70 @@
+import type { BraintreePayPalCheckout } from "../../types/braintree/paypalCheckout";
+import type { BraintreePayPalButtonsComponentProps } from "../../types/braintreePayPalButtonTypes";
+import type { PayPalButtonsComponentProps } from "../../types/paypalButtonTypes";
+
+/**
+ * Override the createOrder callback to send the PayPal checkout instance as argument
+ * to the defined createOrder function for braintree component button
+ *
+ * @param braintreeButtonProps the component button options
+ */
+const decorateCreateOrder = (
+    braintreeButtonProps: BraintreePayPalButtonsComponentProps,
+    payPalCheckoutInstance: BraintreePayPalCheckout
+) => {
+    if (
+        braintreeButtonProps?.createOrder != undefined &&
```

Comment:
Let's validate that it's a function like so:
```suggestion
        typeof braintreeButtonProps.createOrder === 'function'
```

Also, can we remove the `payPalCheckoutInstance` check? We already validate it in BraintreePayPalButtons.tsx